### PR TITLE
Add CreateCardForStandalonePR

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
This should replace old functionality from here https://github.com/SonarSource/sonar-dotnet/blob/master/.github/workflows/CreatePullRequest.yml

`*.ts` and `*.js.map` are autogenerated and do not need review

It does what I want it to do:
Usage: https://github.com/pavel-mikula-sonarsource/GitHubActionPlayground/blob/master/.github/workflows/CreatePullRequest.yml
Test run: https://github.com/pavel-mikula-sonarsource/GitHubActionPlayground/actions/runs/3409800998/jobs/5672009138
Result: https://github.com/pavel-mikula-sonarsource/GitHubActionPlayground/pull/42 - PR was added to project

### CODE REVIEW

File: CreateCardForStandalonePR/CreateCardForStandalonePR.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { WebhookPayload } from "@actions/github/lib/interfaces";
+import { OctokitAction } from "../lib/OctokitAction";
+
+class CreateCardForStandalonePR extends OctokitAction {
+    protected async execute(): Promise<void> {
+        const pr = this.payload.pull_request as WebhookPayload["pull_request"] & { id: number };
+        const matches = pr.body == null ? null : pr.body.match(/(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s*#\d+/gi);
```
Comment (ilia-kebets-sonarsource) [inline CreateCardForStandalonePR/CreateCardForStandalonePR.ts:7]: you can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) like:

```javascript
const matches = pr.body?.match(/(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved)\s*#\d+/gi);
```

File: CreateCardForStandalonePR/CreateCardForStandalonePR.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        if (matches) {
+            this.log("Skip, contains '" + matches[0] + "'");
```
Comment (ilia-kebets-sonarsource) [inline CreateCardForStandalonePR/CreateCardForStandalonePR.ts:9]: It's recommended to use backquotes for strings.

File: lib/OctokitAction.ts [hunk 3]
```diff
@@ -14,16 +15,47 @@ export abstract class OctokitAction extends Action {
...
+    protected async createCardIssue(issue: { id: number }, column_id: number): Promise<components["schemas"]["project-card"]> {
+        return this.createCard(column_id, issue.id, "Issue");
+    }
+
+    protected async createCardPullRequest(pr: { id: number }, column_id: number): Promise<components["schemas"]["project-card"]> {
+        return this.createCard(column_id, pr.id, "PullRequest");
+    }
+
+    private async createCard(column_id: number, content_id: number, content_type: string): Promise<components["schemas"]["project-card"]> {
+        if (column_id == 0) {
```
Comment (ilia-kebets-sonarsource) [inline lib/OctokitAction.ts:54]: `===`

**Review by ilia-kebets-sonarsource — APPROVED** [review]
I haven't checked it does what it means to, because I'm not familiar with GH's API, but I wrote some remarks about the JS/TS. (edit `have`->`haven't`)

File: CreateCardForStandalonePR/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  CreateCardForStandalonePR_job:
+    name: Assign PR to the author and create a Kanban card
+    runs-on: ubuntu-latest
+    # Single quotes must be used here https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#literals
+    # PRs from forks don't have required token authorization
+    # Dependabot works directly under our repository, but doesn't have enough priviledges to create project card
+    if: |
+        github.event.pull_request.head.repo.full_name == github.repository
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply CreateCardForStandalonePR/ReadMe.md:36]: This is not a JS, it's a special GitHub YML markup and === doesn't work here:
https://github.com/pavel-mikula-sonarsource/GitHubActionPlayground/actions/runs/3411859082

[the remark this answers is not in the document]
Comment (ilia-kebets-sonarsource) [reply CreateCardForStandalonePR/ReadMe.md:36]: 👍

**Pull request merged**